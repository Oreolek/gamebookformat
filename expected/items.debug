BEGIN DEBUG OUTPUT
Book title: Items
Number of sections: 11


Turn to 1 to begin.
1 (start) -  Demonstrating how to manage player [COLLECT item]Inventory[/COLLECT]. You start the book carrying a [ADD item]sword[/ADD] and a [ADD item]shield[/ADD]. Turn to 2. 
2 (tjunction) -  You have reached a t-junction. Here you find a [FOUND item]key[/FOUND] and a [FOUND item]stick[/FOUND]. You can go west to 11, or east to 7. 
3 (attempt_break_door_with_sword) -  OK. The door is broken, but so is the [DROP item]sword[/DROP]. Turn to 6. 
4 (treasure) -  You found [FOUND item]something valuable[/FOUND], but there is no way forward, so you head back to 2. 
5 (cut_rope) -  OK. That was fun. Turn to 9. 
6 (inside) -  There is a rope here that can be cut using a [HAS item]sword[/HAS]. If you have one and want to do that, see 5. Otherwise turn to 9. 
7 (curse) -  There is a [FOUND item]cursed bracelet[/FOUND] here. You can go on to 10 or go back to 2. You can also drop the [HAS item]stick[/HAS] for no particular reason if you have it, see 8. 
8 (drop_stick) -  OK [DROP item]stick[/DROP] dropped. Turn back to 2 to confirm stick can not be picked up again even if the text says it is there (books work that way, although ideally this dynamic version should provide some hints that it is no longer there). 
9 (won) -  Congratulations, you won.
10 (portal) -  A magic portal ahead will only allow you to pass if you did not pick up the [HAS NOT item]cursed bracelet[/HAS NOT], leading you to 4. If you have the [HAS item]cursed bracelet[/HAS] you have to go back to 2 instead. Actually feel free to head back to 2 either way. 
11 (door) -  There is a locked door here. If you have a [HAS item]key[/HAS] you can use that to open the door, see 6. Being right before the link should be enough for the formatter to figure out that the key is required to be allowed to follow the link. Else you can try to open with the [HAS item]sword[/HAS], if you have it, see 3. Hopefully the magic is good enough to pair pre-conditions to links, or more markup must be added later. You could also try to go back to pick up the key, see 2. 
END DEBUG OUTPUT
