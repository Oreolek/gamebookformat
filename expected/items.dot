digraph gamebook {


1->2
2->7

2->4
3->10
4->2

4->6

4->11
5->2
6->5

6->2
7->8

7->9

7->2
8->10

8->3
9->8
 11->2
}
