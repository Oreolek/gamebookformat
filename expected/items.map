{"drop_stick": 8, "curse": 9, "door": 11, "start": 1, "autotest": 2, "inside": 5, "treasure": 6, "cut_rope": 4, "tjunction": 12, "won": 3, "portal": 7, "attempt_break_door_with_sword": 10}