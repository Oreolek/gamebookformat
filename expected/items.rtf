{\rtf1\ansi\ansicpg1252\cocoartf1038\cocoasubrtf360
{\fonttbl\f0\fswiss\fcharset0 Helvetica;}
{\colortbl;\red255\green255\blue255;}
\paperw11900\paperh16840\margl1440\margr1440\vieww14140\viewh14860\viewkind0
\pard\tx566\tx1133\tx1700\tx2267\tx2834\tx3401\tx3968\tx4535\tx5102\tx5669\tx6236\tx6803\ql\qnatural\pardirnatural

\f0\b\fs24 \cf0

\b \qc Items
\b0\

\b Turn to 1 to begin.
\b0\
\
\b \qc 1
\b0\
\ql  Demonstrating how to manage player Inventory. You start the book carrying a sword and a shield. Turn to \b 6
\b0
.  \
\

\b \qc 2
\b0\
\ql  OK. That was fun. Turn to \b 8
\b0
.  \
\

\b \qc 3
\b0\
\ql  There is a rope here that can be cut using a sword. If you have one and want to do that, see \b 2
\b0
. Otherwise turn to \b 8
\b0
.  \
\

\b \qc 4
\b0\
\ql  There is a locked door here. If you have a key you can use that to open the door, see \b 3
\b0
. Being right before the link should be enough for the formatter to figure out that the key is required to be allowed to follow the link. Else you can try to open with the sword, if you have it, see \b 5
\b0
. Hopefully the magic is good enough to pair pre-conditions to links, or more markup must be added later. You could also try to go back to pick up the key, see \b 6
\b0
.  \
\

\b \qc 5
\b0\
\ql  OK. The door is broken, but so is the sword. Turn to \b 3
\b0
.  \
\

\b \qc 6
\b0\
\ql  You have reached a t-junction. Here you find a key and a stick. You can go west to \b 4
\b0
, or east to \b 10
\b0
.  \
\

\b \qc 7
\b0\
\ql  A magic portal ahead will only allow you to pass if you did not pick up the cursed bracelet, leading you to \b 11
\b0
. If you have the cursed bracelet you have to go back to \b 6
\b0
 instead. Actually feel free to head back to \b 6
\b0
 either way.  \
\

\b \qc 8
\b0\
\ql  Congratulations, you won. \
\

\b \qc 9
\b0\
\ql  OK stick dropped. Turn back to \b 6
\b0
 to confirm stick can not be picked up again even if the text says it is there (books work that way, although ideally this dynamic version should provide some hints that it is no longer there).  \
\

\b \qc 10
\b0\
\ql  There is a cursed bracelet here. You can go on to \b 7
\b0
 or go back to \b 6
\b0
. You can also drop the stick for no particular reason if you have it, see \b 9
\b0
.  \
\

\b \qc 11
\b0\
\ql  You found something valuable, but there is no way forward, so you head back to \b 6
\b0
.  \
\

}
