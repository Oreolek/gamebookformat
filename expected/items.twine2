<tw-storydata name="Items" startnode="1" creator="Twine" creator-version="2.0" format="Harlowe" options="">
<style role="stylesheet" id="twine-user-stylesheet" type="text/twine-css"></style><script role="script" id="twine-user-script" type="text/twine-javascript"></script>
<tw-passagedata pid="1" name="1" tags="" position="10,10"> Demonstrating how to manage player <b>Inventory</b>. You start the book carrying a <b>sword</b> and a <b>shield</b>. Turn to [[12]]
 or go to the autotest at [[2]]
. </tw-passagedata>
<tw-passagedata pid="2" name="2" tags="" position="20,20"> If you have the <b>stick</b> then you must turn immediately to <b>stick</b>[[12]]
, and everything else in this section is ignored. You find a <b>book</b> here. If you have the <b>sword</b> you must turn immediately to <b>sword</b>[[5]]
, again ignoring everything that follows. You drop the <b>something valuable</b> here if you have it, then go on to [[7]]
.</tw-passagedata>
<tw-passagedata pid="3" name="3" tags="" position="30,30"> Congratulations, you won.</tw-passagedata>
<tw-passagedata pid="4" name="4" tags="" position="40,40"> OK. That was fun. Turn to [[3]]
. </tw-passagedata>
<tw-passagedata pid="5" name="5" tags="" position="50,50"> There is a rope here that can be cut using a <b>sword</b>. If you have one and want to do that, see [[4]]
. You can also run to the autotest at [[2]]
. Otherwise turn to [[3]]
. </tw-passagedata>
<tw-passagedata pid="6" name="6" tags="" position="60,60"> You found <b>something valuable</b>, but there is no way forward, so you head back to [[12]]
. </tw-passagedata>
<tw-passagedata pid="7" name="7" tags="" position="70,70"> A magic portal ahead will only allow you to pass if you did not pick up the <b>cursed bracelet</b>, leading you to [[6]]
. If you have the <b>cursed bracelet</b> you have to go back to [[12]]
 instead. Actually feel free to head back to [[12]]
 either way. </tw-passagedata>
<tw-passagedata pid="8" name="8" tags="" position="80,80"> OK <b>stick</b> dropped. Turn back to [[12]]
 to confirm stick can not be picked up again even if the text says it is there (books work that way, although ideally this dynamic version should provide some hints that it is no longer there). </tw-passagedata>
<tw-passagedata pid="9" name="9" tags="" position="90,90"> There is a <b>cursed bracelet</b> here. You can go on to [[7]]
 or go back to [[12]]
. You can also drop the <b>stick</b> for no particular reason if you have it, see [[8]]
. </tw-passagedata>
<tw-passagedata pid="10" name="10" tags="" position="100,100"> OK. The door is broken, but so is the <b>sword</b>. Turn to [[5]]
. </tw-passagedata>
<tw-passagedata pid="11" name="11" tags="" position="110,110"> There is a locked door here. If you have a <b>key</b> you can use that to open the door, see [[5]]
. Being right before the link should be enough for the formatter to figure out that the key is required to be allowed to follow the link. Else you can try to open with the <b>sword</b>, if you have it, see [[10]]
. Hopefully the magic is good enough to pair pre-conditions to links, or more markup must be added later. You could also try to go back to pick up the key, see [[12]]
. </tw-passagedata>
<tw-passagedata pid="12" name="12" tags="" position="120,120"> You have reached a t-junction. Here you find a <b>key</b> and a <b>stick</b>. You can go west to [[11]]
, or east to [[9]]
. </tw-passagedata>
</tw-storydata>
