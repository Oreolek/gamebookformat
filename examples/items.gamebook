* 1 start
Demonstrating how to manage player inventory. You start
the book carrying a [take]sword[/take] and a [take]shield[/take].
Turn to [[continue]].

* continue
Here you find a [found]key[/found]. Go on to [[door]].

* door
There is a locked door here.
If you have a [carries]key[/carries] you can use that to
open the door, see [[inside]]. Being right before the link
should be enough for the formatter to figure out that the
key is required to be allowed to follow the link.
Else you can try to open with the [carries]sword[/carries],
if you have it, see [[attempt_break_door_with_sword]].
Hopefully the magic is good enough to pair
pre-conditions to links, or more markup must be added later.
You could also try to go back to pick up the key, see [[continue]].

* attempt_break_door_with_sword
OK. The door is broken, but so is the [drop]sword[/drop].
Turn to [[inside]].

* inside
There is a rope here that can be cut using a [carries]sword[/carries].
If you have one and want to do that, see [[cut_rope]].
Otherwise turn to [[won]].

* cut_rope
OK. That was fun. Turn to [[won]].

* won
Congratulations, you won.