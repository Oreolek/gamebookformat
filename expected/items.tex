\documentclass[a5,onecolumn]{book}

\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage[hidelinks]{hyperref}
\usepackage{graphicx}

\usepackage[top=3.3cm, bottom=3.3cm, left=2cm, right=2cm]{geometry}
\newif\ifpdf
\ifx\pdfoutput\undefined
  \pdffalse
\else
  \ifnum\pdfoutput=1
    \pdftrue
  \else
    \pdffalse
  \fi
\fi

\title{Items}
\author{Pelle Nilsson}
\date{}

\newcounter{sectionnr}

\begin{document}

\maketitle

\thispagestyle{empty}

\pagestyle{empty}

\clearpage



Turn to 1 to begin.

\phantomsection
\refstepcounter{sectionnr}
\label{section1}
\subsection*{\begin{center} \textbf{1} \end{center}}

 \noindent
 Demonstrating how to manage player Inventory. You start the book carrying a \textbf{sword} and a \textbf{shield}. Turn to \textbf{\autoref{section2}}. 
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section2}
\subsection*{\begin{center} \textbf{2} \end{center}}

 \noindent
 You have reached a t-junction. Here you find a \textbf{key} and a \textbf{stick}. You can go west to \textbf{\autoref{section7}}, or east to \textbf{\autoref{section4}}. 
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section3}
\subsection*{\begin{center} \textbf{3} \end{center}}

 \noindent
 OK. That was fun. Turn to \textbf{\autoref{section10}}. 
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section4}
\subsection*{\begin{center} \textbf{4} \end{center}}

 \noindent
 There is a \textbf{cursed bracelet} here. You can go on to \textbf{\autoref{section6}} or go back to \textbf{\autoref{section2}}. You can also drop the \textit{stick} for no particular reason if you have it, see \textbf{\autoref{section11}}. 
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section5}
\subsection*{\begin{center} \textbf{5} \end{center}}

 \noindent
 You found \textbf{something valuable}, but there is no way forward, so you head back to \textbf{\autoref{section2}}. 
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section6}
\subsection*{\begin{center} \textbf{6} \end{center}}

 \noindent
 A magic portal ahead will only allow you to pass if you did not pick up the \textit{cursed bracelet}, leading you to \textbf{\autoref{section5}}. If you have the \textit{cursed bracelet} you have to go back to \textbf{\autoref{section2}} instead. Actually feel free to head back to \textbf{\autoref{section2}} either way. 
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section7}
\subsection*{\begin{center} \textbf{7} \end{center}}

 \noindent
 There is a locked door here.eh If you have a \textit{key} you can use that to open the door, see \textbf{\autoref{section8}}. Being right before the link should be enough for the formatter to figure out that the key is required to be allowed to follow the link. Else you can try to open with the \textit{sword}, if you have it, see \textbf{\autoref{section9}}. Hopefully the magic is good enough to pair pre-conditions to links, or more markup must be added later. You could also try to go back to pick up the key, see \textbf{\autoref{section2}}. 
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section8}
\subsection*{\begin{center} \textbf{8} \end{center}}

 \noindent
 There is a rope here that can be cut using a \textit{sword}. If you have one and want to do that, see \textbf{\autoref{section3}}. Otherwise turn to \textbf{\autoref{section10}}. 
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section9}
\subsection*{\begin{center} \textbf{9} \end{center}}

 \noindent
 OK. The door is broken, but so is the \textit{sword}
. Turn to \textbf{\autoref{section8}}. 
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section10}
\subsection*{\begin{center} \textbf{10} \end{center}}

 \noindent
 Congratulations, you won.
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section11}
\subsection*{\begin{center} \textbf{11} \end{center}}

 \noindent
 OK \textit{stick}
 dropped. Turn back to \textbf{\autoref{section2}} to confirm stick can not be picked up again even if the text says it is there (books work that way, although ideally this dynamic version should provide some hints that it is no longer there). 
\vspace{1em}
\end{document}
